kitsune/search/v2/tests/signals/test_wiki.py

```diff
@@ -48,27 +49,39 @@ def test_topics_change(self):
         self.assertEqual(None, self.get_doc().topic_ids)
 
     def test_document_delete(self):
+        RevisionFactory(document=self.document, is_approved=True)
         self.document.delete()
 
         with self.assertRaises(NotFoundError):
             self.get_doc()
 
     def test_revision_delete(self):
-        revision = RevisionFactory(document=self.document, keywords="foobar", is_approved=True)
-        revision.delete()
+        RevisionFactory(document=self.document, keywords="revision1", is_approved=True)
+        revision2 = RevisionFactory(document=self.document, keywords="revision2", is_approved=True)
+        self.assertEqual(self.get_doc().keywords["en-US"], "revision2")
+        revision2.delete()
 
-        self.assertNotIn("foobar", self.get_doc().keywords["en-US"])
+        self.assertNotIn("revision2", self.get_doc().keywords["en-US"])
+        self.assertEqual(self.get_doc().keywords["en-US"], "revision1")
 
     def test_product_delete(self):
+        RevisionFactory(document=self.document, is_approved=True)
         product = ProductFactory()
         self.document.products.add(product)
         product.delete()
 
         self.assertEqual(self.get_doc().product_ids, [])
 
     def test_topic_delete(self):
+        RevisionFactory(document=self.document, is_approved=True)
         topic = TopicFactory()
         self.document.topics.add(topic)
         topic.delete()
 
         self.assertEqual(self.get_doc().topic_ids, [])
+
+    def test_non_approved_revision_update(self):
```

Comment:
I might make this a little more specific, maybe `test_document_without_approved_revision`?